staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt.go

```diff
@@ -0,0 +1,242 @@
+	flowSchemas FlowSchemaSeq
+
+	// priorityLevelStates maps the PriorityLevelConfiguration object
+	// name to the state for that level
+	priorityLevelStates map[string]*PriorityLevelState
+}
+
+// FlowSchemaSeq holds sorted set of pointers to FlowSchema objects.
+// FLowSchemaSeq implements `sort.Interface` (TODO: implement this).
+type FlowSchemaSeq []*rmtypesv1a1.FlowSchema
+
+type PriorityLevelState struct {
+	// config holds the configuration after defaulting logic has been applied
+	config rmtypesv1a1.PriorityLevelConfigurationSpec
+
+	// concurrencyLimit is the limit on number executing
+	concurrencyLimit int
+
+	fqs FairQueuingSystem
+}
+
+// requestManagement holds all the state and infrastructure of this
+// filter
+type requestManagement struct {
+	fairQueuingFactory FairQueuingFactory
+
+	// configQueue holds TypedConfigObjectReference values, identifying
+	// config objects that need to be processed
+	configQueue workqueue.RateLimitingInterface
+
+	// numConfigWorkers is the configured number of workers for the config queue
+	numConfigWorkers int
+
+	// plInformer is the informer for priority level config objects
+	plInformer cache.SharedIndexInformer
+
+	// fsInformer is the informer for flow schema config objects
+	fsInformer cache.SharedIndexInformer
+
+	// plLister belongs here too
+	// fsLister belongs here too
+
+	// serverConcurrencyLimit is the limit on the server's total
+	// number of non-exempt requests being served at once.  This comes
+	// from server configuration.
+	serverConcurrencyLimit int
+
+	// requestWaitLimit comes from server configuration.
+	requestWaitLimit time.Duration
+
+	// curState holds a pointer to the current RMState.  That is,
+	// `Load()` produces a `*RMState`.  When a config work queue worker
+	// processes a configuration change, it stores a new pointer here ---
+	// it does NOT side-effect the old `RMState` value.  The new `RMState`
+	// has a freshly constructed slice of FlowSchema pointers and a
+	// freshly constructed map of priority level states.  But the new
+	// `RMState.priorityLevelStates` includes in its range at least all the
+	// `*PriorityLevelState` values of the old `RMState.priorityLevelStates`.
+	// Consequently the filter can load a `*RMState` and work with it
+	// without concern for concurrent updates.  When a priority level is
+	// finally deleted, this will also involve storing a new `*RMState`
+	// pointer here, but in this case the range of the
+	// `RMState.priorityLevels` will be reduced --- by removal of the
+	// priority level that is no longer in use.
+	curState atomic.Value
+}
+
+// TypedConfigObjectReference is a reference to a relevant config API object.
+// No namespace is needed because none of these objects is namespaced.
+type TypedConfigObjectReference struct {
+	Kind string
+	Name string
+}
+
+func (tr *TypedConfigObjectReference) String() string {
+	return tr.Kind + "/" + tr.Name
+}
+
+var reqMgmt *requestManagement
+
+var rmStartOnce sync.Once
```

Comment:
We don't have globals in kube.  We have instances attached to the structs we use to launch.

Reply:
fixed.